pandas/core/arrays/string_.py

```diff
@@ -274,8 +275,21 @@ def astype(self, dtype, copy=True):
         return super().astype(dtype, copy)
 
     def _reduce(self, name, skipna=True, **kwargs):
+        if name in ["min", "max"]:
+            return getattr(self, name)(skipna=skipna, **kwargs)
+
         raise TypeError(f"Cannot perform reduction '{name}' with string dtype")
 
+    def min(self, axis=None, out=None, keepdims=False, skipna=True):
+        nv.validate_min((), dict(out=out, keepdims=keepdims))
+        result = nanops.nanmin(self._ndarray, axis=axis, skipna=skipna)
+        return libmissing.NA if isna(result) else result
+
+    def max(self, axis=None, out=None, keepdims=False, skipna=True):
```

Comment:
If we add the numpy-compat keywords (axis, out, keepdims), we should also test this (or otherwise leave it out for now).

Also, if we keep them, would it be possible to move them into `**kwargs` ? That keeps the signature cleaner (clearer what is actually supported and not just ignored/validated)

Reply:
Ok, I was largely just mimicking the signature from PandasArray min / max

Reply:
If we remove the numpy arguments, I'm assuming we also don't need `nv.validate_min` and `nv.validate_max`?

Reply:
> If we remove the numpy arguments, I'm assuming we also don't need nv.validate_min and nv.validate_max?

Yes, in that case, that isn't needed. 

Now, maybe others prefer to keep the compatibility. But I personally think we can better tackle that separately later

Reply:
That works for me, I'd also be in favor of removing the arguments if they'd never be used, but curious how others feel

Reply:
So to be clear: they are used if you do `np.min(string_array)` (that's what this numpy compatibility is doing). However, if we support this: 1) I would move those keywords into `**kwargs` to give them less visibility (you can't use them, it's just to avoid raising an error when doing np.min(..)), 2) we should also validate axis to be None or 0, 3) we should test this.

So for those reasons, I'd rather do it in a separate PR (but you can do it here as well, it might just generate some extra discussion ;))

Reply:
Yeah I like the `kwargs` idea, and I think a separate PR would help streamline / focus the discussion a bit

Reply:
so we already do this for series (accept the numpy args *only* as kwargs), a followon PR would be fine